### PR TITLE
Try to do CI without the openforms test docker images

### LINKED ISSUE #4: Set up coverage reporting

### LINKED ISSUE #3389: Remove test docker images
### Thema / Theme

Other

### Omschrijving / Description

Currently, OF provides Docker images with the test dependencies installed, so that the extensions modules can run the tests.

One of the issues with this approach is that the Docker images don't contain the files inside the `tests` directories (since these are specified in the `.dockerignore`).

One of the suggested approaches would be to use git submodules to do the testing of the custom extensions.

- [x] Haal centraal HR extension
- [x] Token exchange extension
- [x] Demo extension
- [x] Documentation

This is needed for #3185

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes https://github.com/open-formulieren/open-forms/issues/3389

Fixes #4

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -17,20 +17,68 @@ jobs:
...
+          export OPEN_FORMS_EXTENSIONS=prefill_haalcentraalhr
+
+          python src/manage.py compilemessages
+          coverage run \
+          --concurrency=multiprocessing \
+          --parallel-mode \
+          src/manage.py test prefill_haalcentraalhr \
+            --parallel 2 \
+            --verbosity 2
+          coverage combine
```
Comment (sergei-maertens) [inline .github/workflows/ci.yaml]: Do we need the concurrency here? I reckon you're not running a massive test suite here, so probably we can simplify it with just `coverage run src/manage.py test prefill_haalcentraalhr`?

And is the coverage data actually being uploaded?

File: setup.cfg [hunk 2]
```diff
@@ -56,8 +56,8 @@ tests =
     isort
     black
     flake8
+    coverage
 pep8 = flake8
-coverage = pytest-cov
```
Comment (sergei-maertens) [inline setup.cfg:60]: perhaps we should try to run tests with `pytest` at some point, if we're developing this as a library anyway?

**Review by sergei-maertens — APPROVED** [review]
Just some thoughts, this looks nice. I'd also update our example/demo repository with the new setup.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (SilviaAmAm) [reply .github/workflows/ci.yaml]: It's true, no need for the concurrency for the 3 tests :smile: 

Huh, I saw the codecov report at some point, but now it looks like it didn't upload :thinking:  I'll look into it!

**Pull request merged**